Update the changelog when creating an RC

Gudahtt opened the pull request:
The changelog will now be automatically updated when a release branch is created. A new release header along with changelog entries for any new commits will be added.

Note that this changelog will still need to be manually cleaned up, but it's one less manual step at least.

The old Bash script for adding a new release header to the changelog has been removed, as that functionality is now built into the changelog update script.

A new script has been added to commit any changes made to the manifest and changelog. This step used to happen at the end of the bump manifest version script, but now the changelog update relies upon the manifest version bump happening first, so it needed to be re-ordered. The changes should only be committed on the first run of the branch, as it's contingent upon the manifest changing (due to the version bump). Further changelog updates won't trigger new automatic commits.

Relates to #10752

Manual testing steps:  
  - Create a new release branch
  - See that it updates the changelog automatically

Gudahtt committed:
Update the changelog when creating an RC

The changelog will now be automatically updated when a release branch
is created. A new release header along with changelog entries for any
new commits will be added.

Note that this changelog will still need to be manually cleaned up, but
it's one less manual step at least.

The old Bash script for adding a new release header to the changelog
has been removed, as that functionality is now built into the changelog
update script.

A new script has been added to commit any changes made to the manifest
and changelog. This step used to happen at the end of the bump manifest
version script, but now the changelog update relies upon the manifest
version bump happening first, so it needed to be re-ordered. The
changes should only be committed on the first run of the branch, as
it's contingent upon the manifest changing (due to the version bump).
Further changelog updates won't trigger new automatic commits.

Gudahtt commented:
This depends upon #10794

rekmarks commented on .circleci/scripts/release-commit-verison-bump.sh:
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
```
Typo in file name: `verison` -> `version`.

It appears to be correctly spelled in the CircleCI config, but we should review all references.

Gudahtt replied:
Oops :facepalm: Good catch! Fixed.

rekmarks reviewed (changes requested):
Looks good except for a typo.

Gudahtt committed:
Fix typo in script filename

rekmarks reviewed (dismissed):
LGTM!

Gudahtt commented:
I've just created the branch `Version-v9.3.1` to test this, just to double check that it all works :eyes:

Gudahtt committed:
Make the new script executable

rekmarks commented:
ffffuuuuuuu
```
@lavamoat/allow-scripts automatically updating configuration

removing unneeded configuration for packages:
- fsevents
Done in 2.65s.
allow-scripts configuration requires updates
```

Gudahtt committed:
Fix check for manifest changes

The check for manifest changes was wrong on two counts. Firstly it was
supposed to check that there *are* changes, but instead it was bailing
if there were *no* changes (the opposite of what was intended). Second,
it was including changelog changes as well, when it should have only
considered manifest changes.

rekmarks commented on .circleci/scripts/release-commit-version-bump.sh:
```diff
@@ -18,7 +18,7 @@ fi
 printf '%s\n' 'Commit the manifest version and changelog if the manifest has changed'
 
-if ! git diff --quiet CHANGELOG.md app/manifest/_base.json;
+if git diff --quiet app/manifest/_base.json;
```
😓

Gudahtt commented:
It worked! https://github.com/MetaMask/metamask-extension/commit/d6b5abdc68eece5ed281600900698bc694633767

... after fixing a couple more mistakes :sweat_smile: . I put each bug fix in a separate commit - check out the most recent two since your approval to see what they were.

Gudahtt commented:
That status check shouldn't affect this PR thankfully, because this isn't a release branch. It was only triggered because this commit was used in a release branch. GitHub applies these status checks by commit it seems, not by branch.

rekmarks reviewed (approved):
Alright, now it looks good!
